### PR TITLE
Issue #9498: add code example of AST for TokenTypes.LITERAL_THIS

### PR DESCRIPTION
**Pull request by CryptoSingh1337**
Closes: #9498
image

**Successfully Compiled**
image

**Test.java**
`λ cat Test.java`
```
class TestClass {

        private int id;
        private String name;

        public TestClass() {
                this(1, "NULL");
        }

        public TestClass(int id, String name) {
                this.id = id;
                this.name = name;
        }
}
```

**AST**
`λ java -jar checkstyle-8.41.1-all.jar -t Test.java`
```
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|--LITERAL_CLASS -> class [1:0]
|--IDENT -> TestClass [1:6]
`--OBJBLOCK -> OBJBLOCK [1:16]
    |--LCURLY -> { [1:16]
    |--VARIABLE_DEF -> VARIABLE_DEF [3:1]
    |   |--MODIFIERS -> MODIFIERS [3:1]
    |   |   `--LITERAL_PRIVATE -> private [3:1]
    |   |--TYPE -> TYPE [3:9]
    |   |   `--LITERAL_INT -> int [3:9]
    |   |--IDENT -> id [3:13]
    |   `--SEMI -> ; [3:15]
    |--VARIABLE_DEF -> VARIABLE_DEF [4:1]
    |   |--MODIFIERS -> MODIFIERS [4:1]
    |   |   `--LITERAL_PRIVATE -> private [4:1]
    |   |--TYPE -> TYPE [4:9]
    |   |   `--IDENT -> String [4:9]
    |   |--IDENT -> name [4:16]
    |   `--SEMI -> ; [4:20]
    |--CTOR_DEF -> CTOR_DEF [6:1]
    |   |--MODIFIERS -> MODIFIERS [6:1]
    |   |   `--LITERAL_PUBLIC -> public [6:1]
    |   |--IDENT -> TestClass [6:8]
    |   |--LPAREN -> ( [6:17]
    |   |--PARAMETERS -> PARAMETERS [6:18]
    |   |--RPAREN -> ) [6:18]
    |   `--SLIST -> { [6:20]
    |       |--CTOR_CALL -> this [7:2]
    |       |   |--LPAREN -> ( [7:6]
    |       |   |--ELIST -> ELIST [7:7]
    |       |   |   |--EXPR -> EXPR [7:7]
    |       |   |   |   `--NUM_INT -> 1 [7:7]
    |       |   |   |--COMMA -> , [7:8]
    |       |   |   `--EXPR -> EXPR [7:10]
    |       |   |       `--STRING_LITERAL -> "NULL" [7:10]
    |       |   |--RPAREN -> ) [7:16]
    |       |   `--SEMI -> ; [7:17]
    |       `--RCURLY -> } [8:1]
    |--CTOR_DEF -> CTOR_DEF [10:1]
    |   |--MODIFIERS -> MODIFIERS [10:1]
    |   |   `--LITERAL_PUBLIC -> public [10:1]
    |   |--IDENT -> TestClass [10:8]
    |   |--LPAREN -> ( [10:17]
    |   |--PARAMETERS -> PARAMETERS [10:18]
    |   |   |--PARAMETER_DEF -> PARAMETER_DEF [10:18]
    |   |   |   |--MODIFIERS -> MODIFIERS [10:18]
    |   |   |   |--TYPE -> TYPE [10:18]
    |   |   |   |   `--LITERAL_INT -> int [10:18]
    |   |   |   `--IDENT -> id [10:22]
    |   |   |--COMMA -> , [10:24]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [10:26]
    |   |       |--MODIFIERS -> MODIFIERS [10:26]
    |   |       |--TYPE -> TYPE [10:26]
    |   |       |   `--IDENT -> String [10:26]
    |   |       `--IDENT -> name [10:33]
    |   |--RPAREN -> ) [10:37]
    |   `--SLIST -> { [10:39]
    |       |--EXPR -> EXPR [11:10]
    |       |   `--ASSIGN -> = [11:10]
    |       |       |--DOT -> . [11:6]
    |       |       |   |--LITERAL_THIS -> this [11:2]
    |       |       |   `--IDENT -> id [11:7]
    |       |       `--IDENT -> id [11:12]
    |       |--SEMI -> ; [11:14]
    |       |--EXPR -> EXPR [12:12]
    |       |   `--ASSIGN -> = [12:12]
    |       |       |--DOT -> . [12:6]
    |       |       |   |--LITERAL_THIS -> this [12:2]
    |       |       |   `--IDENT -> name [12:7]
    |       |       `--IDENT -> name [12:14]
    |       |--SEMI -> ; [12:18]
    |       `--RCURLY -> } [13:1]
    `--RCURLY -> } [14:0]
```

```
# Printing the code that we care
λ java -jar checkstyle-8.41.1-all.jar -t Test.java | grep "7:\|12:"
```
```
    |       |--CTOR_CALL -> this [7:2]
    |       |   |--LPAREN -> ( [7:6]
    |       |   |--ELIST -> ELIST [7:7]
    |       |   |   |--EXPR -> EXPR [7:7]
    |       |   |   |   `--NUM_INT -> 1 [7:7]
    |       |   |   |--COMMA -> , [7:8]
    |       |   |   `--EXPR -> EXPR [7:10]
    |       |   |       `--STRING_LITERAL -> "NULL" [7:10]
    |       |   |--RPAREN -> ) [7:16]
    |       |   `--SEMI -> ; [7:17]
    |       |--EXPR -> EXPR [12:12]
    |       |   `--ASSIGN -> = [12:12]
    |       |       |--DOT -> . [12:6]
    |       |       |   |--LITERAL_THIS -> this [12:2]
    |       |       |   `--IDENT -> name [12:7]
    |       |       `--IDENT -> name [12:14]
    |       |--SEMI -> ; [12:18]
```

### Expected update for javadoc
```
 EXPR -> EXPR
  `--ASSIGN -> =
      |--DOT -> .
      |   |--LITERAL_THIS -> this
      |   `--IDENT -> name
      `--IDENT -> name
 SEMI -> ;

 CTOR_CALL -> this
  |--LPAREN -> (
  |--ELIST -> ELIST
  |   |--EXPR -> EXPR
  |   |   `--NUM_INT -> 1
  |   |--COMMA -> ,
  |   `--EXPR -> EXPR
  |       `--STRING_LITERAL -> "NULL"
  |--RPAREN -> )
  `--SEMI -> ;
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -2059,7 +2059,40 @@
...
+     * The {@code this} keyword use to refer the current object.
+     * This can also be used to call the constructor.
+     *
+     * <p>For example:</p>
+     * <pre>
+     * this.name = name;
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * |--EXPR -&gt; EXPR
```
Comment (AkMo3) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2071]: remove `|` it is not required as it not lead to anywhere.

Comment (CryptoSingh1337) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2071]: this is connected with `SEMI`

Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2071]: Please see https://github.com/checkstyle/checkstyle/pull/9899

Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Resolved.

**Review by romani — APPROVED** [review]

**Pull request merged**